helix-term/src/commands/lsp.rs

```diff
@@ -1254,6 +1254,25 @@ pub fn compute_inlay_hints_for_all_views(editor: &mut Editor, jobs: &mut crate::
     }
 }
 
+fn lsp_annotations_area(view: &View, doc: &Document) -> (usize, usize) {
+    // Compute ~3 times the current view heigh, that way some scrolling
```

Comment:
Maybe 2/2.5 times is a better option? Just to be on the more conservative side, since we don't know exactly what kind document the user will open (maybe it is just full of colour definitions, many per line)

Reply:
Doing 0.5-1 screens worth of extra computation is not that expensive in terms of performance but its better for user experience rather if we suddenly see a bunch of these color swatches appear

This exact number was previously used for inlay hints too

---

Hmm, I'm thinking about this and: We can see 1 screen at once. Ctrl + f will move us down an entire screen. Ctrl + b will move up an entire screen

If we render 3 screens view height, then pressing Ctrl + f will move us to the bottom screen and then quickly render the 4th screen. If it was something like 2 or 2.5 screens, we could see an area of these swatches not being rendered at all for a second

Reply:
Makes sense, thanks for walking me through your line of thought!